dubbo-common/src/main/java/org/apache/dubbo/common/config/CompositeConfiguration.java

```diff
@@ -21,19 +21,20 @@
 import org.apache.dubbo.common.utils.ArrayUtils;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
 
 /**
  * This is an abstraction specially customized for the sequence Dubbo retrieves properties.
  */
 public class CompositeConfiguration implements Configuration {
-    private Logger logger = LoggerFactory.getLogger(CompositeConfiguration.class);
+    private final Logger logger = LoggerFactory.getLogger(CompositeConfiguration.class);
 
     /**
      * List holding all the configuration
      */
-    private List<Configuration> configList = new LinkedList<>();
+    private final List<Configuration> configList = Collections.synchronizedList(new LinkedList<>());
```

Comment:
Use CopyOnWriteArrayList may be better. `synchronizedList` will block some read operations.

Reply:
@AlbumenJ done, thanks